.github/workflows/ci.yml

```diff
@@ -115,11 +108,10 @@ jobs:
 
       - name: Validate package build
         run: |
-          source .venv/bin/activate
           for dir in packages/*/; do uv build "$dir" --out-dir dist; done
 
       - name: Publish Package
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
```

Comment:
`upload-artifact@v3` will stop working in a week ([see the announcement](https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/)) but for some reason even today  some runs of the CI for this PR would fail with a message saying that `upload-artifact@v3` is not supported.